projects/node-http-exercise/node-http-exercise.md

```diff
@@ -0,0 +1,255 @@
+
+5. Create the `/echo?message=xyz` route, which should respond with 200 and the message as the body. For this one, to help you out, you should start by console-logging the `req` object to find out how to pull out the message. Also, as a hint, the solution uses a check before the helper function also for that one.
+
+6. Create the `/error` route, which should should respond with 500.
+
+7. Create the `/time.json` route, which should should respond with 200, body with Content-Type of `application/json` and `{ "now": "Sat Feb 15 2020 14:29:20 GMT-0800" }` where it's the current time of the response.
+
+8. Create the `/github` route, which should respond with 302 that redirects to [Techtonica GitHub page](https://github.com/Techtonica).
+
+### Questions to Consider
+
+- Why do you think it is better practice not to hard-code everything within the response object and header?
+- How is this API construct different than a RESTful API?
+- In terms of security, what do you think the implementation with the different checks for routes either in the object or earlier certain endpoints yield? Could this be improved?
+
+### Hints
+
+- You can abruptly end a connection by using `res.end()` and `server.close()`.
+- If called as such, the `Date()` created will look like this: `Wed Oct 14 2020 21:22:50 GMT-0400 (Eastern Daylight Time)`, but for this exercise, we do not want what's in the parentheses at the end. For any more information, I invite you to check the [official Date documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date).
+- If you want to make a check before going into the helper function, you can first create a variable with the route, including the slash, then check if the `pathname` triple-equals to that variable. Then you can either cut the connection shortly or do anything else before your regular routes checks. For example, you could use the `res.end()` and `server.close()` or inquire on the `path.query.message` to get more information on the url.
+
+### Testing
+
+1. If you haven't already done it, [download and install Postman](https://www.postman.com/product/api-client/).
+1. Import the `node-http-server-assignment.postman_collection.json` file that is located in the `node-http-exercise` folder.
+1. When your server is running, simply choose the route you want to test in the left pane and click on `SEND`. You should then see the response and the status code.
+
+### Extensions
+
+- If you want, instead of always restarting the server when you make a change, you can install `Nodemon` by doing `npm i nodemon` within your project folder. For this, you need to have `npm` installed and will be changing the start script in the `package.json`. If you are interested in doing so, here is a [very good tutorial](https://www.digitalocean.com/community/tutorials/workflow-nodemon).
+- If you want cleaner code, use destructuring in the given helper function.
```

Comment:
I would replace this extension with the lookup idea if we think it is useful to introduce at this stage.

Reply:
fixed in [799330c](https://github.com/Techtonica/curriculum/pull/1406/commits/799330cf2b82a0e0d74d3cd2e258f97adc3e70db)